### PR TITLE
Update makefile to match our CI

### PR DESCRIPTION
**Pull request by VersusFacit**
### Description

Having these new recipes brings local testing in compliance with our CI testing commands. This helps users fulfill the checkbox below about running local tests that CI will run.

Fun fact: this was adapted out a script I used to debug structured log problems related to #5496 

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,13 +65,20 @@ test: .env ## Runs unit tests with py and code checks against staged changes.
...
-	$(DOCKER_CMD) tox -e py-integration -- -nauto
+	$(COND_CI_FLAGS) $(DOCKER_CMD) tox -e py-integration -- -nauto
 
 .PHONY: integration-fail-fast
 integration-fail-fast: .env ## Runs postgres integration tests with py-integration in "fail fast" mode.
 	@\
 	$(DOCKER_CMD) tox -e py-integration -- -x -nauto
 
+.PHONY: structured-log-test
+structured-log-test: clean
```
Comment (nathaniel-may) [inline Makefile:76]: I'm inclined to name this "interop" instead. I know structured logs are the only tests in there now, but in theory interop is a layer of abstraction above integration tests. unit (blackbox=function) < integration (blackbox=function+warehouse) < interop (blackbox=dbt+warehouse).

**Review by nathaniel-may — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -6,6 +6,23 @@ ifeq ($(USE_DOCKER),true)
...
+# log testing.
+CI_VARS =\
+	DBT_TEST_USER_1=dbt_test_user_1\
+	DBT_TEST_USER_2=dbt_test_user_2\
+	DBT_TEST_USER_3=dbt_test_user_3\
+	RUSTFLAGS="-D warnings"\
+	LOG_DIR=./logs\
+	DBT_LOG_FORMAT=json
+
+ifeq ($(USE_CI_FLAGS),true)
```
[the remark this answers is not in the document]
Comment (VersusFacit) [reply Makefile:22]: I don't think `if` is a part of the makefile language.

https://www.gnu.org/software/make/manual/html_node/Conditional-Syntax.html

File: Makefile (hunk 1 above)
Comment (VersusFacit) [reply Makefile:76]: Sure! Can do

File: Makefile [hunk 3]
```diff
@@ -6,6 +6,23 @@ ifeq ($(USE_DOCKER),true)
 	DOCKER_CMD := docker-compose run --rm test
 endif
 
+LOGS_DIR := ./logs
+
+# Optional flag to invoke tests using our CI env.
+# But we always want these active for structured
+# log testing.
+CI_VARS =\
```
[the remark this answers is not in the document]
Comment (VersusFacit) [reply Makefile:14]: I thought drawing the distinction might be nice but you're right--it's over partitioning of the closure.

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (nathaniel-may) [reply Makefile:22]: it is: https://www.gnu.org/software/make/manual/html_node/Conditional-Functions.html

File: Makefile [hunk 4]
```diff
@@ -48,13 +61,20 @@ test: .env ## Runs unit tests with py and code checks against staged changes.
 .PHONY: integration
 integration: .env ## Runs postgres integration tests with py-integration
 	@\
-	$(DOCKER_CMD) tox -e py-integration -- -nauto
+	$(if $(USE_CI_FLAGS), $(CI_FLAGS)) $(DOCKER_CMD) tox -e py-integration -- -nauto
```
Comment (nathaniel-may) [inline Makefile:64]: oh this reads so nicely now. much better.

**Review by nathaniel-may — APPROVED** [review]
LGTM

File: Makefile (hunk 4 above)
Comment (VersusFacit) [reply Makefile:64]: I really, really like this!!

**Pull request merged**